pkg/cloudprovider/nodecontroller/nodecontroller.go

```diff
@@ -567,6 +340,9 @@ func (nc *NodeController) monitorNodeStatus() error {
 	if err != nil {
 		return err
 	}
+	if nc.allocateNodeCIDRs {
+		nc.reconcileNodeCIDRs(nodes)
```

Comment:
// TODO: use stuff from the controller framework to watch, to reduce the number of lists executed.